From: Tim Waugh <twaugh at redhat.com>
Date: Mon, 10 Aug 2009 13:55:40 +0000 (+0100)
Subject: Clean up temporary file and fix memory leak (trac #12).
X-Git-Url: http://git.fedorahosted.org/git/?p=patchutils.git;a=commitdiff_plain;h=c4f3c09d729905eca186226b1eb3a9394bc8b0bc

Clean up temporary file and fix memory leak (trac #12).

* src/util.c (xtmpfile): Clean up temporary file (trac #12).
Also free the memory allocated to hold its name.
---

diff --git a/ChangeLog b/ChangeLog
index df7b9bb..db0fae4 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2009-08-10  Tim Waugh  <twaugh@redhat.com>
+
+	* src/util.c (xtmpfile): Clean up temporary file (trac #12).
+	Also free the memory allocated to hold its name.
+
 2009-01-23  Tim Waugh  <twaugh@redhat.com>
 
 	* configure.in: Version 0.3.1.
diff --git a/src/util.c b/src/util.c
index 371fc19..9910fc7 100644
--- a/src/util.c
+++ b/src/util.c
@@ -111,6 +111,8 @@ FILE *xtmpfile (void)
 	ret = fdopen (fd, "w+b");
 	if (ret == NULL)
 		error (EXIT_FAILURE, errno, "fdopen");
+	unlink (tmpfname);
+	free (tmpfname);
 	return ret;
 }
 
